contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1339 @@
+        operator.nuInTStake -= _amount - tRemainder;
+
+        emit Unstaked(_operator, _amount - tRemainder);
+    }
+
+    /// @notice Sets cached legacy stake amount to 0, sets the liquid T stake
+    ///         amount to 0 and withdraws all liquid T from the stake to the
+    ///         owner. Reverts if there is at least one non-zero authorization.
+    ///         Can be called only by the delegation owner and operator.
+    function unstakeAll(address _operator)
+        external
+        override
+        onlyOwnerOrOperator(_operator)
+    {
+        OperatorInfo storage operator = operators[_operator];
+        require(operator.owner != address(0), "Operator has no stake");
+        require(
+            operator.authorizedApplications.length == 0,
+            "At least one application is still authorized"
+        );
+        require(
+            operator.tStake == 0 ||
+                minTStakeAmount == 0 ||
+                operator.startTStakingTimestamp + MIN_STAKE_TIME <=
+                /* solhint-disable-next-line not-rely-on-time */
+                block.timestamp,
+            "Unstaking is possible only after 24 hours"
+        );
+
+        emit Unstaked(
+            _operator,
+            operator.tStake + operator.keepInTStake + operator.nuInTStake
+        );
+        uint96 amount = operator.tStake;
+        operator.tStake = 0;
+        operator.keepInTStake = 0;
+        operator.nuInTStake = 0;
+        if (amount > 0) {
+            token.safeTransfer(operator.owner, amount);
+        }
+    }
+
+    //
+    //
+    // Keeping information in sync
+    //
+    //
+
+    /// @notice Notifies about the discrepancy between legacy KEEP active stake
+    ///         and the amount cached in T staking contract. Slashes the operator
+    ///         in case the amount cached is higher than the actual active stake
+    ///         amount in KEEP staking contract. Needs to update authorizations
+    ///         of all affected applications and execute an involuntary
+    ///         authorization decrease on all affected applications. Can be called
+    ///         by anyone, notifier receives a reward.
+    function notifyKeepStakeDiscrepancy(address _operator) external override {
+        OperatorInfo storage operator = operators[_operator];
+        require(operator.keepInTStake > 0, "Nothing to slash");
+
+        (uint256 keepStakeAmount, , uint256 undelegatedAt) = keepStakingContract
+            .getDelegationInfo(_operator);
+        (uint96 tAmount, ) = keepToT(keepStakeAmount);
+
+        require(
+            operator.keepInTStake > tAmount || undelegatedAt != 0,
+            "There is no discrepancy"
+        );
+        operator.keepInTStake = tAmount;
+        seizeKeep(
+            operator,
+            _operator,
+            stakeDiscrepancyPenalty,
+            stakeDiscrepancyRewardMultiplier
+        );
+
+        emit TokensSeized(_operator, tAmount - operator.keepInTStake);
```

Comment:
Do we need a Discrepancy event as well?

Reply:
Added flag to `TokenSeized`, it's pretty cheap comparing with additional event
✔️